cmake/common/versionconfig.cmake

```diff
@@ -33,23 +33,22 @@ endif()
 
 # Set beta/rc versions if suffix included in version string
 if(_obs_version MATCHES "[0-9]+\\.[0-9]+\\.[0-9]+-rc[0-9]+")
-  string(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)-rc([0-9]+).*$" "\\1;\\2;\\3;\\4" _obs_release_candidate
-                       ${_obs_version})
+  string(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)-rc([0-9]+).*$" "\\4" _obs_release_candidate ${_obs_version})
 elseif(_obs_version MATCHES "[0-9]+\\.[0-9]+\\.[0-9]+-beta[0-9]+")
-  string(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)-beta([0-9]+).*$" "\\1;\\2;\\3;\\4" _obs_beta ${_obs_version})
+  string(REGEX REPLACE "([0-9]+)\\.([0-9]+)\\.([0-9]+)-beta([0-9]+).*$" "\\4" _obs_beta ${_obs_version})
+endif()
+
+# Set commit for untagged version comparisons in the Windows updater
+if(OS_WINDOWS)
```

Comment:
No OS-Specific code during bootstrapping (only exception is `osconfig` itself) - I don't think there's any harm in setting it by default.

Though if it is _only_ used in the Windows updater, that code should be moved inside that CMakeLists file instead.

Reply:
Alright, I'll just change it to work like all the other variables.

Reply:
The value seems only used in `UI/cmake/os-windows.cmake`, so the regex could be moved there. It shouldn't have any side-effects here, but it feels iffy adding this globally if it's really only used by a single module for a specific OS for a specific compilation unit.